### DISCUSSION TITLE
Feature suggestion: download files from older windows builds?

### DISCUSSION THREAD

**Discussion by Anixx** [valinet/ExplorerPatcher#2208] [Ideas]
In light that the Windows 10 taskbar is going to be removed from Windows, I suggest to implement a function that would be able to download the missing Windows components from older Windows releases in automatic manner.

Besides the possible future necessity to download the older version of Explorer for Windows 10 taskbar, there  are also other Windows complonents that can be replaced with those from older Windows versions. Downloading and replacing them manually is boring, especially because one needs to account for MUI files and take ownership. Also, the original files need to be replaced and this change may be reverted on each update. EP on the other hand may implement on-the-fly redirection without replacing the original files in the system folders.

These are some components that could be restored to the ones from older Windows versions:

* The non-immersive UAC prompt: https://winclassic.net/thread/1778/non-xaml-uac-back
* The old Task Manager and System Configuration (msconfig.exe) dialog: https://winaero.com/windows-10-classic-task-manager-for-windows-11/
* Some Control Panel applets: https://winclassic.net/thread/1779/restoring-control-panel-applets-windows
* LogonUI screen: https://winclassic.net/thread/1804/windows-logon-ui-1507-1511 (Win 8.1 version); https://winclassic.net/thread/154/windows-10-console-logon-screen (console logon)
* Non-immersive Calculator, Paint (Win7 style and Vista style) https://www.askvg.com/how-to-get-the-good-old-ms-paint-without-ribbons-working-in-windows-7/ , WordPad (also, Win7 style and Vista style) https://www.askvg.com/how-to-get-the-good-old-wordpad-without-ribbons-working-in-windows-7/
* Windows Mail https://www.eightforums.com/threads/how-to-reinstate-winmail-onto-windows-8-10-11.3698/ (quite complicated procedure to do it manually!)

**Comment by pyrates999** [comment]
Doing that could then break a windows 10/11 install with every update that is released.  They would need to do this before updating:

Uninstall EP, reboot.  Then run this to repair windows:

```sfc /scannow
dism /online /cleanup-image /scanhealth
dism /online /cleanup-image /restorehealth
dism /online /cleanup-image /checkhealth
dism /online /cleanup-image /analyzecomponentstore
dism /online /cleanup-image /startcomponentcleanup```

Then install the update.

Then install EP and reapply all the settings.
